.github/workflows/protocol_tests.yml

```diff
@@ -97,8 +114,28 @@ jobs:
       - name: Run Everything just in case something was missed
         # can't use gas limit because some setUp function use more than the limit
         # Excludes integration tests, because they require an anvil devchain running on the localhost.
-        run: forge test -vvv --no-match-contract RegistryIntegrationTest
+        run: |
+          forge test -vvv \
+          --no-match-contract RegistryIntegrationTest
+      
+      - name: Generate migrations and run devchain
+        if: success() || failure()
+        run: ./migrations_sol/create_and_migrate_anvil_devchain.sh
+      
+      - name: Run integration tests against devchain
+        if: success() || failure()
+        run: |
+          forge test -vvv \
+          --match-path "test-sol/integration/*" \
+          --fork-url http://127.0.0.1:$ANVIL_PORT
       
-      - name: Generate migrations
+      - name: Run e2e tests against devchain
         if: success() || failure()
-        run: ./migrations_sol/run_integration_tests_in_anvil.sh
+        run: |
+          forge test -vvv \
+          --match-path "test-sol/e2e/*" \
+          --fork-url http://127.0.0.1:$ANVIL_PORT
```

Comment:
Note-to-self: This line won't work until I merge this PR into my branch: 
- https://github.com/celo-org/celo-monorepo/pull/11020

I'm waiting for that PR to be reviewed.

Reply:
That's done in https://github.com/celo-org/celo-monorepo/pull/11023/commits/77ea5cc29246f63878ba56729ac6e5d10d71a833